frontend/server/src/Controllers/User.php

```diff
@@ -3703,7 +3703,13 @@ private static function getSelectedFilter(
         }
         return ['filteredBy' => null, 'value' => null];
     }
-
+    public static function getSupportedProgrammingLanguages(): array {
+        return array_filter(
+            \OmegaUp\Controllers\Run::SUPPORTED_LANGUAGES,
+            fn($key) => $key !== 'cat',
+            ARRAY_FILTER_USE_KEY
+        );
+    }
```

Comment:
For PHP, the array type declaration you added is correct, but let's remember that PHP is a weakly typed language. So, to have greater control over inputs and outputs in functions, we use a tool to add types. This tool is PSALM, so we should be more specific when declaring the type. In this case, it is an array with string keys and string values, which is declared as follows:

```suggestion

    /**
     * @return array<string,string>
     */
    public static function getSupportedProgrammingLanguages(): array {
        return array_filter(
            \OmegaUp\Controllers\Run::SUPPORTED_LANGUAGES,
            fn($key) => $key !== 'cat',
            ARRAY_FILTER_USE_KEY
        );
    }

```